sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala

```diff
@@ -1174,10 +1249,12 @@ class CodegenContext extends Logging {
       // Currently, we will do this for all non-leaf only expression trees (i.e. expr trees with
       // at least two nodes) as the cost of doing it is expected to be low.
 
+      val subExprCode = s"${addNewFunction(fnName, fn)}($INPUT_ROW);"
       subexprFunctions += s"${addNewFunction(fnName, fn)}($INPUT_ROW);"
```

Comment:
nit: shall we use `subexprFunctions += subExprCode` here? otherwise we are calling `addNewFunction` twice.

Reply:
Oh yes, as the functions in class is a map, it will overwrite. But yes, we should use `subExprCode`. Let me submit a followup.

Reply:
#33305